src/Compilers/CSharp/Test/Emit2/CodeGen/CodeGenRefReadonlyParameterTests.cs

```diff
@@ -0,0 +1,474 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests.CodeGen;
+
+public class CodeGenRefReadonlyParameterTests : CSharpTestBase
+{
+    private const string RequiresLocationAttributeName = "RequiresLocationAttribute";
+
+    [Fact]
+    public void Method()
+    {
+        var source = """
+            class C
+            {
+                public void M(ref readonly int p) { }
+            }
+            """;
+        var verifier = CompileAndVerify(source, targetFramework: TargetFramework.NetStandard20,
+            sourceSymbolValidator: verify, symbolValidator: verify);
+        verifier.VerifyDiagnostics();
+        verifier.VerifyTypeIL("C", """
```

Comment:
The use of `VerifyTypeIL` is okay I suppose, but it's verbose given that you're only trying to verify a couple of things: 1. `[in]` flag and `RequiresLocation` attribute emitted on parameter, 2. type of parameter. Consider a more focused/compact test.

Reply:
I agree it's verbose. It would be enough to verify just the IL signature. Not sure how to do that, though.